### DISCUSSION TITLE
Validating input types with Authorized

### DISCUSSION THREAD

**Discussion by RobinKa** [MichalLytek/typegraphql-prisma#292] [Q&A] [answered]
I'm using an autogenerated schema and am trying to validate input type fields for a mutation with `Authorized`, but no auth check actually happens. Using eg. `MinLength` instead does work though, so I'm wondering if I misunderstood something or whether this is a bug.

Schema:
```gql
input AggregatorCreateInput {
  name: String!
  ...
}

type Mutation {
  createAggregator(data: AggregatorCreateInput!): Aggregator!
  ...
}
```

ValidateNested / Authorized, does not work:
```ts
applyInputTypesEnhanceMap({
  AggregatorCreateInput: {
    fields: {
      name: [Authorized()],
    },
  },
})

applyArgsTypesEnhanceMap({
  CreateAggregatorArgs: {
    fields: {
      data: [ValidateNested()]
    },
  },
})

```
ValidateNested / MinLength, does work:
```ts
applyInputTypesEnhanceMap({
  AggregatorCreateInput: {
    fields: {
      name: [MinLength(10)],
    },
  },
})

applyArgsTypesEnhanceMap({
  CreateAggregatorArgs: {
    fields: {
      data: [ValidateNested()]
    },
  },
})
```

**Comment by MichalLytek** [comment, marked as answer]
Have you configured the auth layer according to main docs?

**Reply by RobinKa** [reply]
I passed an auth checker to `buildSchemaSync` which does not get called in the example above. `Authorized` does work when I use it eg. with model or resolver enhance maps and the auth checker gets called.

**Reply by MichalLytek** [reply]
So please check the docs again - it's stated there that the autorization does not work for input types. It only works for resolvers (query, mutation, field resolver) because they have some executable code. Args and inputs are only shape of data, there's no transform logic for them. Only class-validator works because it checks the whole nested object at the same time.
